constructor_examples/erc721_token_constructor.py

```diff
@@ -0,0 +1,1195 @@
+ */
+contract BurnableToken is ERC721BasicToken {
+    function burn(uint256 _tokenId) public {
+        _burn(msg.sender, _tokenId);
+    }
+}
+
+
+contract MintableToken is ERC721Token, Ownable {
+    event MintFinished();
+
+    bool public mintingFinished = false;
+
+    modifier canMint() {
+        require(!mintingFinished);
+        _;
+    }
+    
+    function mint(address _to, uint256 _tokenId) public onlyOwner canMint {
+        _mint(_to, _tokenId);
+    }
+    
+    /**
+     * @dev Function to stop minting new tokens.
+     * @return True if the operation was successful.
+     */
+    function finishMinting() onlyOwner canMint public returns (bool) {
+        mintingFinished = true;
+        MintFinished();
+        return true;
+    }    
+}
+
+/**
+ * @title Capped token
+ * @dev Mintable token with a token cap.
+ */
+contract CappedToken is MintableToken {
+    uint256 public cap;
+
+    function CappedToken(uint256 _cap) public {
+        require(_cap > 0);
+        cap = _cap;
+    }
+
+    /**
+     * @dev Function to mint tokens
+     * @param _to The address that will receive the minted tokens.
+     * @param _tokenId id of the new token
+     * @return A boolean that indicates if the operation was successful.
+     */
+    function mint(address _to, uint256 _tokenId) onlyOwner canMint public {
+        require(totalSupply().add(1) <= cap);
+
+        return super.mint(_to, _tokenId);
+    }
+
+}
+
+/**
+ * @title Pausable
+ * @dev Base contract which allows children to implement an emergency stop mechanism.
+ */
+contract Pausable is Ownable {
+    event Pause();
+    event Unpause();
+
+    bool public paused = false;
+
+    /**
+     * @dev Modifier to make a function callable only when the contract is not paused.
+     */
+    modifier whenNotPaused() {
+        require(!paused);
+        _;
+    }
+
+    /**
+     * @dev Modifier to make a function callable only when the contract is paused.
+     */
+    modifier whenPaused() {
+        require(paused);
+        _;
+    }
+
+    /**
+     * @dev called by the owner to pause, triggers stopped state
+     */
+    function pause() onlyOwner whenNotPaused public {
+        paused = true;
+        Pause();
+    }
+
+    /**
+     * @dev called by the owner to unpause, returns to normal state
+     */
+    function unpause() onlyOwner whenPaused public {
+        paused = false;
+        Unpause();
+    }
+}
+
+/**
+ * @title Pausable token
+ * @dev StandardToken modified with pausable transfers.
+ **/
+contract PausableToken is ERC721BasicToken, Pausable {
+    function approve(address _to, uint256 _tokenId) public whenNotPaused {
+        return super.approve(_to, _tokenId);
+    }
+
+    function setApprovalForAll(address _to, bool _approved) public whenNotPaused {
+        return super.setApprovalForAll(_to, _approved);
+    }
+
+    function transferFrom(address _from, address _to, uint256 _tokenId) public whenNotPaused canTransfer(_tokenId) {
```

Comment:
it is not necessary to use modifier canTransfer(_tokenId) since it be checked in parent contract

+2 fns below